mongodbatlas/organizations.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2019 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const (
+	orgsBasePath = "orgs"
+)
+
+// OrganizationsService provides access to the organization related functions in the Atlas API.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/organizations/
+type OrganizationsService interface {
+	List(context.Context, *ListOptions) (*Organizations, *Response, error)
+	Get(context.Context, string) (*Organization, *Response, error)
+	Projects(context.Context, string, *ListOptions) (*Projects, *Response, error)
+	Users(context.Context, string, *ListOptions) (*AtlasUsersResponse, *Response, error)
+	Delete(context.Context, string) (*Response, error)
+}
+
+// OrganizationsServiceOp provides an implementation of the OrganizationsService interface
+type OrganizationsServiceOp service
+
+var _ OrganizationsService = &OrganizationsServiceOp{}
+
+// Organization represents the structure of an organization.
+type Organization struct {
+	ID    string  `json:"id,omitempty"`
+	Links []*Link `json:"links,omitempty"`
+	Name  string  `json:"name,omitempty"`
+}
+
+// Organizations represents an array of organization
+type Organizations struct {
+	Links      []*Link         `json:"links"`
+	Results    []*Organization `json:"results"`
+	TotalCount int             `json:"totalCount"`
+}
+
+// List gets all organizations.
```

Comment:
In Atlas a programmatic API key can only be associated with one org, so all it will ever return is one org.  In fact, looking at our doc: https://docs.atlas.mongodb.com/reference/api/organization-get-all-projects/  I wonder if this is an artifact of when we had user api keys?  So basically, this is just an FYI to see if it makes sense in the context of Atlas.

Reply:
Makes sense but wanted to ensure you went in eyes-wide-open :)  Thanks @gssbzn !